mindsdb/integrations/handlers/salesforce_handler/salesforce_handler.py

```diff
@@ -0,0 +1,199 @@
+                security_token='security_token')
+
+
+class SalesforceHandler(DatabaseHandler):
+    """
+     This handler handles connection and execution of Salesforce statements.
+    """
+
+    name = "salesforce"
+
+    def __init__(self, name: str, connection_data: Optional[dict], **kwargs):
+        """
+        Initialize the handler
+        Args:
+            name (str): name of particular handler instance
+                        connection_data (dict): parameters for connecting to the database
+            **kwargs: arbitrary keyword arguments.
+        """
+        super().__init__(name)
+        self.parser = parse_sql
+        self.dialect = 'salesforce'
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+
+        self.connection = None
+        self.is_connected = False
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self) -> StatusResponse:
+        """
+        Set up the connection required by the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+        if self.is_connected is True:
+            return StatusResponse(True)
+
+        self.connection = sf.client(
+            'salesforce',
+            sf_username=self.connection_data['user'],
+            sf_password=self.connection_data['password'],
+            sf_security_token=self.connection_data['security_token']
+        )
+        self.is_connected = True
+
+        return self.connection
+
+    def disconnect(self):
+        self.connection = None
+        self.is_connected = False
+
+        return self.connection
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check the Salesforce connection
+        """
+        response = StatusResponse(False)
+        need_to_close = self.is_connected is False
+        if need_to_close is True:
+            self.connect()
+        try:
+            self.connection.query("SELECT Id FROM Account LIMIT 1")
+            response = StatusResponse(True)
+        except Exception as e:
+            response = StatusResponse(False, error=str(e))
+        if need_to_close is True:
+            self.disconnect()
+        return response
+
+    def native_query(self, query: str) -> Response:
+        """
+        Receive SOQL query from Salesforce and return the result as a pandas dataframe
+        Args: 
+            query (str): SOQL query
+        Returns:
+            HandlerResponse
+        """
+        need_to_close = self.is_connected is False
+        connection = self.connection
+
+        if need_to_close is True:
+            self.connect()
+            connection = self.connection
+
+        try:
+            result = connection.select_object_content(
+                sf_instance=self.connection_data['sf_instance'],
+                reportId=self.connection_data['reportId'],
+                export='?isdtp=p1&export=1&enc=UTF-8&xf=csv',
```

Comment:
Is this query parameter required `?isdtp=p1&export=1&enc=UTF-8&xf=csv` if not maybe better to remove it?

Reply:
I found it in the salesforce docs - I thought it was to define it as a cvs, but simple_salesforce returns data as JSON, so I can remove that

Thanks for the feedback. I had to take a sick day yesterday, so I'm making these changes now